admin/server/organizations.go

```diff
@@ -1194,3 +1211,40 @@ func whitelistedDomainToPB(a *database.OrganizationWhitelistedDomainWithJoinedRo
 		Role:   a.RoleName,
 	}
 }
+
+func planDowngrade(newPan *billing.Plan, org *database.Organization) bool {
+	// nil or negative values are considered as unlimited
+	if comparableInt(newPan.Quotas.NumProjects) < comparableInt(&org.QuotaProjects) {
+		return true
+	}
+	if comparableInt(newPan.Quotas.NumDeployments) < comparableInt(&org.QuotaDeployments) {
+		return true
+	}
+	if comparableInt(newPan.Quotas.NumSlotsTotal) < comparableInt(&org.QuotaSlotsTotal) {
+		return true
+	}
+	if comparableInt(newPan.Quotas.NumSlotsPerDeployment) < comparableInt(&org.QuotaSlotsPerDeployment) {
+		return true
+	}
+	if comparableInt(newPan.Quotas.NumOutstandingInvites) < comparableInt(&org.QuotaOutstandingInvites) {
+		return true
+	}
+	if comparableInt64(newPan.Quotas.StorageLimitBytesPerDeployment) < comparableInt64(&org.QuotaStorageLimitBytesPerDeployment) {
+		return true
+	}
+	return false
+}
+
+func comparableInt(v *int) int {
+	if v == nil || *v < 0 {
```

Comment:
if the quotas is not set (nil) the we should use the default value instead of unlimited. can just change the plan creation logic to use default value instead of nil in case a quota is not set

Reply:
Not sure, but I think I'm fine with having unlimited when a quota is not explicitly set – it reduces user pain if we mess up, and should help in dev and testing. What do you think?